File: easy/27.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        l = 0
        r = len(nums) - 1
        k = len(nums)
        while l <= r:
            if nums[l] == val:
                nums[l] = nums[r]
                r -= 1
                k -= 1
            else:
                l += 1
        return k

File: easy/27.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = nums.length;
        int l = 0;
        int r = nums.length - 1;
        while (l <= r){
            if (nums[l] == val){
                nums[l] = nums[r];
                --r;
                --k;
            } else {
                ++l;
            }
        }
        return k;
    }
}
